### PR TITLE
Update LS docs index page with subsections

### PR DESCRIPTION
**Pull request by IMS94**
## Purpose
$subject

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: docs/language-server/README.md [hunk 1]
```diff
@@ -11,14 +11,19 @@ These documentations are intended for the developers who wish to understand the
 server implementation and for developers who are willing to write new language server features.
 
 ## Features
-* [Ballerina Language Server Features](Features.md)
+* [Completions and Code Actions](Features.md)
+
+## Document Event Sync PubSub Model
+A mechanism to get notified on
```
Comment (mindula) [inline docs/language-server/README.md:17]: How about we add a comma at the end?
```suggestion
A mechanism to get notified on, 
```

Comment (IMS94) [reply docs/language-server/README.md]: I had forgotten to finish this sentence. Fixed it.

File: docs/language-server/README.md [hunk 2]
```diff
@@ -11,14 +11,20 @@ These documentations are intended for the developers who wish to understand the
 server implementation and for developers who are willing to write new language server features.
 
 ## Features
-* [Ballerina Language Server Features](Features.md)
+* [Completions and Code Actions](Features.md)
+
+## Document Event Sync PubSub Model
+A mechanism to get notified on project update, document change and other interested events across the Language Server.
```
Comment (mindula) [inline docs/language-server/README.md:17]:
```suggestion
A mechanism to get notified on project update, document change and other interesting events across the Language Server.
```

**Review by mindula — APPROVED** [review]
LGTM

**Pull request merged**